packages/ember/tests/routing/router_service_test/transitionTo_test.js

```diff
@@ -364,6 +364,64 @@ moduleFor(
       });
     }
 
+    async ['@test RouterService#transitionTo should not refresh parent models if the parent query params stay the same'](
+      assert
+    ) {
+      let parentModelHookCallCount = 0;
+      this.add(
+        'route:parent',
+        Route.extend({
+          queryParams: {
+            parentParam: {
+              refreshModel: true,
+            },
+          },
+          model() {
+            parentModelHookCallCount++;
+            return 'model';
+          },
+        })
+      );
+      this.add(
+        'route:parent.child',
+        Route.extend({
+          queryParams: {
+            childParam: {
+              refreshModel: true,
+            },
+          },
+        })
+      );
+      this.add(
+        'controller:parent',
+        Controller.extend({
+          parentParam: 'parentParamDefaultValue',
+        })
+      );
+      this.add(
+        'controller:parent.child',
+        Controller.extend({
+          childParam: 'childParamDefaultValue',
+        })
+      );
+
+      await this.visit('/?parentParam=parentParamDefaultValue');
+      assert.equal(parentModelHookCallCount, 1);
+
```

Comment:
I'm what is the this.routerService.currentURL at this point.

Reply:
It's `/?parentParam=parentParamDefaultValue`. I assume that this is because the when transitioning to a support URL, the query params are left alone. But when transitioning using a route name and `queryParams` hash, default values are not included in the resulting URL.